### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by satvikshubham**
Signed-off-by: Satvik Shubham Singh <58393226+satvikshubham@users.noreply.github.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Business Hour Picker
+****
+Business hour picker helps you to pick day and time corresponding to a set of working days and show them separately.  
+## Dependencies
+****
+For using bhlib in your app, add the below dependency in the entry/package.json  
+```
+"dependencies": {
```
Comment (CoolGuy-098765432) [inline README.md:8]: add dependency command like this
```npm i @ohos/bhlib```

File: bhlib/src/main/ets/components/MainPage/Business.ets [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export class Business {
+  private workingDays: boolean[] = [false, false, false, false, false, false, false];
```
Comment (CoolGuy-098765432) [inline bhlib/src/main/ets/components/MainPage/Business.ets]: add properties in custom component using model class

File: bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Business } from './Business';
+
+@CustomDialog
+export struct TimeSelectorDialog {
+  time: string
```
Comment (CoolGuy-098765432) [inline bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets:22]: add properties in custom component using model class

File: bhlib/src/main/ets/components/MainPage/BusinessHoursWeekView.ets [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Business } from './Business';
+
+@Component
+export struct BusinessHoursWeekView {
+  private showBusinessModel: Business = undefined
```
Comment (CoolGuy-098765432) [inline bhlib/src/main/ets/components/MainPage/BusinessHoursWeekView.ets]: add properties in custom component using model class

File: bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+import { BusinessModel } from './BusinessModel';
+
+@CustomDialog
+export struct TimeSelectorDialog {
+  time: string
+  where: string
+  dayInd: number
+  format: number
+  callback: (selIndex, where, dayInd) => void
+  arr12: string [] = [
```
Comment (HMOS-Developer) [inline bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets]: add these arr12 & arr24 variables in separate constant file

File: bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BusinessModel } from './BusinessModel';
+
+@CustomDialog
+export struct TimeSelectorDialog {
+  time: string
+  where: string
```
Comment (HMOS-Developer) [inline bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets:23]: time, where, dayInd, format
add these properties in model class

File: bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+          ForEach(this.arr24, (item) => {
+            ListItem() {
+              Text(item)
+                .fontSize(16)
+                .width('100%')
+                .padding(10)
+                .textAlign(TextAlign.Center)
+                .backgroundColor(0xFFFFFF)
+                .onClick(() => {
+                  this.callback(item, this.where, this.dayInd)
```
Comment (HMOS-Developer) [inline bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets]: handle null & undefined validations on callback variable

File: bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+          ForEach(this.arr12, (item) => {
+            ListItem() {
+              Text(item)
+                .fontSize(16)
+                .width('100%')
+                .padding(10)
+                .textAlign(TextAlign.Center)
+                .backgroundColor(0xFFFFFF)
+                .onClick(() => {
+                  this.callback(item, this.where, this.dayInd)
```
Comment (HMOS-Developer) [inline bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets:62]: handle null & undefined validations on callback variable

File: bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+      this.fTime = selIndex
+      this.business.setFromTime(dayInd, this.fTime)
+    }
+    else {
+      this.tTime = selIndex
+      this.business.setToTime(dayInd, this.tTime)
+    }
+  }
+
+  bhTextColor: string = '#000000'
```
Comment (HMOS-Developer) [inline bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets]: create a model class for BusinessDayAndTime component and add all variables from this component in model class

File: bhlib/src/main/ets/components/MainPage/BusinessHoursWeekView.ets [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BusinessModel } from './BusinessModel';
+
+@Component
+export struct BusinessHoursWeekView {
+  private showBusinessModel: BusinessModel = undefined
+  isToday = new Date().getDay()
```
Comment (HMOS-Developer) [inline bhlib/src/main/ets/components/MainPage/BusinessHoursWeekView.ets:23]: create a model class for BusinessHoursWeekView component and add all variables from this component in model class

File: bhlib/src/main/ets/components/MainPage/BusinessModel.ets [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+  constructor(business?: BusinessModel) {
+    if (business) {
+      this.workingDays = business.workingDays
+      this.from = business.from
+      this.to = business.to
+    }
+  }
+
+  setWorkingDay = (day: number, v: boolean): void => {
+    this.workingDays[day - 1] = v;
```
Comment (HMOS-Developer) [inline bhlib/src/main/ets/components/MainPage/BusinessModel.ets]: handle null & undefined validations before assigning values to a variable in all setter methods

File: bhlib/src/main/ets/components/MainPage/BusinessModel.ets [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+  }
+  setFromTime = (day: number, fromTime: string): void => {
+    this.from[day - 1] = fromTime;
```
Comment (HMOS-Developer) [inline bhlib/src/main/ets/components/MainPage/BusinessModel.ets:35]: handle null & undefined validations before assigning values to a variable

File: bhlib/src/main/ets/components/MainPage/BusinessPicker.ets [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+import { BusinessModel } from './BusinessModel';
+import { BusinessDayAndTime } from './BusinessDayAndTime';
+
+@Component
+export struct BusinessPicker {
+  @Link selectedBusinessModel: BusinessModel
+  day: string [] = BusinessModel.dayArr
+  arr: number[] = [0, 1, 2, 3, 4, 5, 6]
```
Comment (HMOS-Developer) [inline bhlib/src/main/ets/components/MainPage/BusinessPicker.ets:25]: create a model class for BusinessPicker component and add all variables from this component in model class

File: entry/src/main/ets/MainAbility/app.ets [hunk 14]
```diff
@@ -0,0 +1,8 @@
+export default {
```
Comment (HMOS-Developer) [inline entry/src/main/ets/MainAbility/app.ets]: add copyright text here

**Comment by HMOS-Developer** [comment]
Add Copyright text in all ts/ets files

File: bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets (hunk 5 above)
Comment (PatExplores) [reply bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets]: done

File: bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets (hunk 6 above)
Comment (PatExplores) [reply bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets:23]: Done

File: bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets (hunk 8 above)
Comment (PatExplores) [reply bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets:62]: done

File: bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets (hunk 9 above)
Comment (PatExplores) [reply bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets]: done

File: bhlib/src/main/ets/components/MainPage/BusinessHoursWeekView.ets (hunk 10 above)
Comment (PatExplores) [reply bhlib/src/main/ets/components/MainPage/BusinessHoursWeekView.ets:23]: Done

File: bhlib/src/main/ets/components/MainPage/BusinessModel.ets (hunk 11 above)
Comment (PatExplores) [reply bhlib/src/main/ets/components/MainPage/BusinessModel.ets]: Done

File: bhlib/src/main/ets/components/MainPage/BusinessPicker.ets (hunk 13 above)
Comment (PatExplores) [reply bhlib/src/main/ets/components/MainPage/BusinessPicker.ets:25]: Done

File: entry/src/main/ets/MainAbility/app.ets (hunk 14 above)
Comment (PatExplores) [reply entry/src/main/ets/MainAbility/app.ets]: Done

File: bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets [hunk 15]
```diff
@@ -0,0 +1,222 @@
...
+  controller: CustomDialogController
+
+  build() {
+    Column() {
+      List({ initialIndex: 0 }) {
+        if (this.format == 24) {
+          ForEach(this.arr24, (item) => {
+            ListItem() {
+              Text(item)
+                .fontSize(16)
```
Comment (archanaatr) [inline bhlib/src/main/ets/components/MainPage/BusinessDayAndTime.ets]: Mark all fontsizes as fp and dimentions as vp, recheck in all files and handle.

File: package.json [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+    "directoryLevel": "project"
+  },
+  "description": "example description",
+  "repository": {},
+  "license": "ISC",
+  "dependencies": {
+    "hypium": "^1.0.0",
+    "@ohos/hvigor": "1.0.6",
+    "@ohos/hvigor-ohos-plugin": "1.0.6",
+    "@ohos/bhlib": "file:./bhlib"
```
Comment (archanaatr) [inline package.json]: remove the bhlib dependency from here and add at entry's package.json

**Review by archanaatr — COMMENTED** [review]
pls check the comments

**Review by archanaatr — APPROVED** [review]
ok

**Pull request merged**